packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/TableDemo/RowWrapperForTableRowWrapperDemo.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+import { RowWrapperProps } from '@patternfly/react-table';
+import { css } from '@patternfly/react-styles';
+import styles from '@patternfly/react-styles/css/components/Table/table';
+
+export default class CustomRowWrapper extends React.Component<RowWrapperProps, {}> {
+  render() {
+    const {
+      trRef,
+      className,
+      rowProps,
+      row,
+      ...props
+    } = this.props;
+
+    const isEvenOddRow = rowProps && (rowProps.rowIndex) % 2;
```

Comment:
Looks like you're checking if it's odd https://github.com/patternfly/patternfly-react/pull/3160/files#diff-7b522b96301a408012fc2cc357c0f91fR27. Do you think `isEvenOddRow` could possibly be renamed to `isOddRow`? I was confused by the name of it since in this example you're applying the styles to only the odd rows.

Reply:
Also, if you're trying to check for odd rows, you do `(rowProps.rowIndex + 1) % 2` in https://github.com/patternfly/patternfly-react/pull/3160/files#diff-140e5f0b0acc2241453a7c0859691b15R15 so I'm guessing you want to do the same thing here too?

Reply:
and just out of curiosity, what does `rowProps && ...` do to the value of isEvenOddRow? It looks like `rowProps` is declared as an object in `RowWrapper.tsx` so I'm really curious about how the bool expression is evaluated here

Reply:
> and just out of curiosity, what does `rowProps && ...` do to the value of isEvenOddRow? It looks like `rowProps` is declared as an object in `RowWrapper.tsx` so I'm really curious about how the bool expression is evaluated here

I used logical AND here because [rowProps can be undefined](https://github.com/patternfly/patternfly-react/pull/3160/files#diff-89b3fdfa90b01f903ab08494efcf7b4dR19). Without this, copying the example code into a TypeScript based app will immediately throw errors. This happens because we're accessing the rowIndex prop, and in the case where rowProps are undefined, attempting to access `.rowIndex` would throw an error. So it is basically being used as a sort of type guard.

Does that make sense? Probably a lot we can do to improve this, let me know if you see a better way.